pkg/operator/controller/machineconfig/controller.go

```diff
@@ -289,7 +282,6 @@ func (r *MachineConfigReconciler) removeFinalizer(ctx context.Context, req ctrl.
 
 		withoutFinalizers.Finalizers = newFinalizers
 		if err := r.ClientUpdate(ctx, withoutFinalizers); err != nil {
-			r.Log.Error(err, "failed to remove nodeobservabilitymachineconfig resource finalizers")
```

Comment:
Maybe we could add the following for context:
```
err = utilerrors.NewAggregate([]error{err, fmt.Errorf("failed to remove nodeobservabilitymachineconfig resource finalizers: %w", err)}

Reply:
This message contains the duplicate information, the error returned by this method will be wrapped in the error with the same message up the call stack (in `cleanUp` method).